week10/bulletin-board/src/pages/Detail.jsx

```diff
@@ -0,0 +1,352 @@
+import {
+  AddCommentApi,
+  DeleteCommentApi,
+  LoadCommentsApi,
+} from "../api/comment";
+import { useParams, useNavigate } from "react-router-dom";
+import { MdDeleteOutline } from "react-icons/md";
+import styled from "styled-components";
+
+const PostContainer = styled.div`
+  margin-bottom: 2vh;
+  list-style-type: none;
+  padding-left: 3vw;
+  padding-right: 3vw;
+  padding-top: 2vh;
+  padding-bottom: 1vh;
+`;
+
+const ContentTitle = styled.div`
+  font-size: 25px;
+  font-weight: bold;
+  padding-bottom: 0.5;
+`;
+const TitleAndHeart = styled.div`
+  display: flex;
+  flex-direction: column;
+`;
+
+const Hearts = styled.div`
+  display: flex;
+  flex-direction: row;
+  margin-top: 1vh;
+`;
+const HeartButtonsOFF = styled.button`
+  margin-left: 0.8vw;
+  background-color: transparent;
+  border: none;
+  cursor: pointer;
+`;
+const HeartButtonsON = styled.button`
+  background-color: transparent;
+  border: none;
+  cursor: pointer;
+`;
+
+const Likes = styled.div`
+  margin-bottom: 2vh;
+`;
+
+const PostButtons = styled.div`
+  flex-direction: row;
+`;
+const FixAndDel = styled.div``;
+
+const FixButton = styled.button`
+  margin-right: 1vw;
+`;
+
+const DeleteButton = styled.button``;
+
+const CommentsWrapper = styled.div`
+  display: flex;
+  align-items: start;
+  justify-content: start;
+  flex-direction: column;
+  margin-top: 1.5vh;
+`;
+
+const CommentWrite = styled.div`
+  display: flex;
+  flex-direction: row;
+`;
+
+const CommentInput = styled.textarea`
+  border: none;
+  &:focus {
+    outline: none;
+  }
+  height: 45px;
+  width: 86vw;
+  border: 2px solid #ced4da;
+  resize: none;
+  margin-bottom: 1.5vh;
+`;
+
+const CommentUploadButton = styled.button`
+  height: 52px;
+  margin-left: 1vw;
+  width: 5.2vw;
+`;
+
+const CommentBox = styled.div`
+  width: 100%;
+`;
+
+const Comments = styled.div`
+  display: flex;
+  flex-direction: column;
+
+  width: 86vw;
+  align-items: start;
+  justify-content: start;
+`;
+
+const CommentItem = styled.div`
+  display: flex;
+  border-bottom: 2px solid #ced4da;
+  margin-bottom: 1vh;
+  justify-content: space-between;
+  width: 100%;
+`;
+
+const DelIcon = styled(MdDeleteOutline)`
+  font-size: 25px;
+  color: #73787d;
+  cursor: pointer;
+`;
+
+const CommentAuthor = styled.div`
+  font-size: 14px;
+  color: #495057;
+  margin-bottom: 0.5vh;
+  margin-right: 1vw;
+`;
+
+const CommentContent = styled.div`
+  display: flex;
+  align-items: center;
+  width: 100%;
+`;
+
+const InputText = styled.textarea`
+  border: none;
+  &:focus {
+    outline: none;
+  }
+  height: 30vh;
+  width: 50vw;
+  margin-top: 2vh;
+  border: solid 1.5px #ced4da;
+  border-radius: 5px;
+  resize: none;
+`;
+
+function DetailPage() {
+  const navigate = useNavigate();
+  const { postId } = useParams();
+  const [post, setPost] = useState(null);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+
```

Comment:
3. 현재 브라우저에 로그인한 유저(나)의 닉네임 정보를 저장할 state를 생성합니다.

```suggestion
const [myNickname, setMyNickname] = useState("");
```